### PR TITLE
[ESP32] Moved the redundent documentation to common place

### PR DESCRIPTION
**Pull request by shubhamdp**
#### Problem
* Each ESP32 application README contains redundant text
* ESP32 documentation is distributed into multiple readme files so difficult to find useful stuff

#### Change overview
* Remove the redundant text and move most possible stuff to common place `docs/guides/esp32`

#### Testing
Browsed through the parsed files on Github

### CODE REVIEW

**Review by dhrishi — APPROVED** [review]

File: examples/all-clusters-app/esp32/README.md [hunk 1]
```diff
@@ -3,292 +3,22 @@
...
-    -   [Building the Example Application](#building-the-example-application)
-    -   [Commissioning and cluster control](#commissioning-and-cluster-control)
-        -   [Setting up chip-tool](#setting-up-chip-tool)
-        -   [Commissioning over BLE](#commissioning-over-ble)
-        -   [Cluster control](#cluster-control)
-        -   [Flashing app using script](#flashing-app-using-script)
-        -   [Note](#note)
-    -   [Using the RPC console](#using-the-rpc-console)
-    -   [Device Tracing](#device-tracing)
+Refer [setup ESP-IDF and CHIP Environment](#setup-esp-idf-and-chip-environment)
```
Comment (bzbarsky-apple) [inline examples/all-clusters-app/esp32/README.md:6]:
```suggestion
Refer to [setup ESP-IDF and CHIP Environment](#setup-esp-idf-and-chip-environment)
```

And similar in other places.

File: examples/all-clusters-app/esp32/README.md [hunk 2]
```diff
@@ -3,292 +3,22 @@
...
+and
+[Build, Flash, Monitor, and Commissioning](#build-flash-monitor-and-commissioning)
+guide to get started.
```
Comment (bzbarsky-apple) [inline examples/all-clusters-app/esp32/README.md:9]:
```suggestion
guides to get started.
```
and again similar in other places.

**Review by bzbarsky-apple — APPROVED** [review]

File: examples/all-clusters-app/esp32/README.md (hunk 1 above)
Comment (shubhamdp) [reply examples/all-clusters-app/esp32/README.md]: Thanks Boris, i've changed this and others locally.

File: examples/all-clusters-app/esp32/README.md (hunk 2 above)
Comment (shubhamdp) [reply examples/all-clusters-app/esp32/README.md]: thanks, done!

**Pull request merged**